src/ng/compile.js

```diff
@@ -1711,6 +1744,27 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
 
         directiveName = directive.name;
 
+        // If we encounter a condition that can result in transclusion on the directive,
+        // then scan ahead in the remaining directives for others that may cause a multiple
+        // transclusion error to be thrown during the compilation process.  If a matching directive
+        // is found, then we know that when we encounter a transcluded directive, we need to eagerly
+        // compile the `transclude` function rather than doing it lazily in order to throw
+        // exceptions at the correct time
+        if (!didScanForMultipleTransclusion && ((directive.templateUrl && directive.replace)
```

Comment:
It's not, good catch!  That was actually indicative of a missing unit test - I've added one that tests that the template/replace:true combination throws a multiple transclusion error exactly as the templateUrl case does.  As expected, that test failed with my code, so I've checked that case too.